uberfire-nio2-backport/uberfire-nio2-impls/uberfire-nio2-jgit/src/test/java/org/uberfire/java/nio/fs/jgit/JGitFileSystemImplProviderTest.java

```diff
@@ -2163,8 +2152,126 @@ protected void setupFileSystemsManager() {
+            final OutputStream outStream = provider.newOutputStream(path);
+            outStream.write("little baby another-branch".getBytes());
+            outStream.close();
+        }
+
+        final Path source = provider.getPath(URI.create("git://another-branch@movebranch-repo/"));
+        final Path target = provider.getPath(URI.create("git://another-branch-moved@movebranch-repo/"));
+
+        provider.move(source,
+                      target);
+
+        Throwable extractContentCall = catchThrowable(() -> extractContent(provider.getPath(URI.create("git://another-branch@movebranch-repo/dorinha.txt"))));
+
+        assertThat(extractContentCall).isInstanceOf(NoSuchFileException.class);
+
+        final String contentMoved = extractContent(provider.getPath(URI.create("git://another-branch-moved@movebranch-repo/dorinha.txt")));
+
+        assertThat(contentMoved).isNotNull().isEqualTo("little baby another-branch");
+    }
+
+    @Test(expected = IllegalStateException.class)
+    public void moveBranchesNotAtTheSameFSShouldNotBeAllowedTest() throws IOException {
+        final URI newRepo = URI.create("git://movebranch-repo");
+        provider.newFileSystem(newRepo,
+                               EMPTY_ENV);
+
+        final URI anotherRepo = URI.create("git://another-repo");
+        provider.newFileSystem(anotherRepo,
+                               EMPTY_ENV);
+
+        {
+            final Path path = provider.getPath(URI.create("git://another-branch@movebranch-repo/dorinha.txt"));
+
+            final OutputStream outStream = provider.newOutputStream(path);
+            outStream.write("little baby another-branch".getBytes());
+            outStream.close();
+        }
+
+        final Path source = provider.getPath(URI.create("git://another-branch@movebranch-repo/"));
+        final Path target = provider.getPath(URI.create("git://another-branch-moved@another-repo/"));
+
+        provider.move(source,
+                      target);
+    }
+
+    @Test(expected = IllegalStateException.class)
+    public void copyBranchesNotAtTheSameFSShouldNotBeAllowedTest() throws IOException {
+        final URI newRepo = URI.create("git://movebranch-repo");
+        provider.newFileSystem(newRepo,
+                               EMPTY_ENV);
+
+        final URI anotherRepo = URI.create("git://another-repo");
+        provider.newFileSystem(anotherRepo,
+                               EMPTY_ENV);
+
+        {
+            final Path path = provider.getPath(URI.create("git://another-branch@movebranch-repo/dorinha.txt"));
+
+            final OutputStream outStream = provider.newOutputStream(path);
+            outStream.write("little baby another-branch".getBytes());
+            outStream.close();
+        }
+
+        final Path source = provider.getPath(URI.create("git://another-branch@movebranch-repo/"));
+        final Path target = provider.getPath(URI.create("git://another-branch-moved@another-repo/"));
+
+        provider.copy(source,
```

Comment:
I'd recommend using `assertThatThrownBy()` from assertj instead of Junit's expected exception. See plenty of examples [here](https://github.com/kiegroup/appformer/pull/219/files. This has the advantage of being precise about which statement should throw the exception AND enables you to check exception message.